File: backend/schemas/comment.py
from pydantic import BaseModel
from datetime import datetime
from schemas.user import UserOut


class CommentCreate(BaseModel):
    content: str
    parent_comment_id: int | None = None


class CommentOut(BaseModel):
    id: int
    content: str
    status: str
    created_at: datetime
    user: UserOut
    parent_comment_id: int | None
    review_id: int

    model_config = {"from_attributes": True}

File: frontend/src/types/comment.ts
import type { User } from './auth'

export interface Comment {
  id: number
  content: string
  status: string
  created_at: string
  user: User
  parent_comment_id: number | null
  review_id: number
}

export interface CommentNode extends Comment {
  replies: CommentNode[]
}
